EIPS/eip-6239.md

```diff
@@ -0,0 +1,167 @@
+
+```Solidity
+
+interface ISemanticSBTUpdate{
+    /**
+     * @notice Update the RDF data for Semantic Soulbound Token. Implementors can assign updaters as needed, for example, the token issuer.
+     * @dev Emits the `UpdateRDF` event.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfData RDF data is a collection of RDF statements that are used to represent information about resources.
+     */
+    function updateRDF(uint256 tokenId, RDFData memory rdfData) external;
+
+}
+
+```
+
+## Rationale
+
+### Method Specification
+
+rdfOf (uint256 tokenId): Query the RDF data for the Semantic Soulbound Token by tokenId. The returned RDF data format conforms to the W3C RDF standard. RDF data is a collection of RDF statements that are used to represent information about resources. An RDF statement, also known as a triple, is a unit of information in the RDF data model. It consists of three parts: a subject, a predicate, and an object. 
+
+updateRDF (uint256 tokenId, RDFData rdfData): This **OPTIONAL** method is used when it needs to update the RDF data for Semantic SBT. Use this method to find the RDF data for Semantic SBT by tokenId and perform the update. The input RDF data **MUST** conform to W3C RDF standards. When implementing this method, **SHALL** assign updaters as needed, for example, the token issuer can be assigned as an updater. When calling this method, the UpdateRDF event **MUST** be triggered to notify the listener for performing relevant business update.
+
+schemaURI(): This **OPTIONAL** method is used to query the URIs of the schema for the RDF data. RDF Schema is an extension of the basic RDF vocabulary and provides a data-modelling vocabulary for RDF data. It is **RECOMMENDED** to store the RDF Schema in decentralized storage such as Arweave or IPFS. The URIs are then stored in the contract and can be queried by this method.
+
+### Event Specification
+
+CreateRDF: When minting a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform operations with the created RDF data. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object. 
+
+UpdateRDF: When updating RDF data for a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform update operations accordingly with the updated RDF data. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object. 
+
+RemoveRDF: When burning or revoking a Semantic Soulbound Token, this event **MUST** be triggered to notify the listener to perform operations with the removed RDF data for the Semantic SBT. When calling the event, the input RDF data **MUST** be RDF statements, which are units of information consisting of three parts: a subject, a predicate, and an object. 
+
```

Comment:
All of this seems like it should belong in the Specification section. The Specification section should describe the externally visible behaviour of implementations, while the Rationale section must describe individual technical choices made within the EIP itself.

Reply:
Thank you for the reminder. We have moved this section into the Specification section and re-written the Rationale.

Reply:
Changed as suggested.